app/models/concerns/rubygem_searchable.rb

```diff
@@ -32,17 +32,31 @@ def as_indexed_json(_options = {})
     settings number_of_shards: 1,
              number_of_replicas: 1,
              analysis: {
-               analyzer: {
-                 rubygem: {
+               filter: {
+                 autocomplete_ngram: {
+                   max_gram: 20,
+                   min_gram: 2,
+                   type: 'edge_ngram'
+                 }
+               },
+               tokenizer: {
+                 special_characters: {
                    type: 'pattern',
                    pattern: "[\s#{Regexp.escape(SPECIAL_CHARACTERS)}]+"
                  }
+               },
+               analyzer: {
+                 autocomplete_index: {
+                   type: 'custom',
+                   tokenizer: 'special_characters',
+                   filter: %w(lowercase autocomplete_ngram)
+                 }
                }
              }
 
     mapping do
       indexes :name, type: 'multi_field' do
-        indexes :name, analyzer: 'rubygem'
+        indexes :name, index_analyzer: 'autocomplete_index', search_analyzer: 'whitespace'
```

Comment:
`index_analyzer` is [removed in ES 2.0
](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking_20_mapping_changes.html#_analyzer_mappings)

I understand that contribution.md does not specify ES version, but I would advocate that we use the latest ES 2.3. ES 2 has so many improvements.